optuna/integration/wandb.py

```diff
@@ -126,10 +147,43 @@ def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -
         metrics = {name: value for name, value in zip(names, trial.values)}
         attributes = {"direction": [d.name for d in study.directions]}
 
-        wandb.config.update(attributes)
-        wandb.log({**trial.params, **metrics}, step=trial.number)
+        step = trial.number if wandb.run else None
+        run = wandb.run
 
-    def _initialize_run(self) -> None:
-        """Initializes Weights & Biases run."""
+        # When an user doesn't use `track_in_wandb` decorator.
+        # Might create an extra run if an user logs in wandb but doesn't use the decorator.
+
+        if not run:
+            run = self._initialize_run()
+            run.name = f"trial/{trial.number}/{run.name}"
 
-        wandb.init(**self._wandb_kwargs)
+        run.log({**trial.params, **metrics}, step=step)
+
+        run.config.update({**attributes})
+
+        if self._as_multirun:
+            run.finish()
```

Comment:
How about this?
```suggestion
            run.tags = tuple(self._wandb_kwargs.get("tags", ())) + (study.study_name, )
            run.finish()
```

Reply:
That would have worked if he had access to the `study` object 🙂. Is there way to get access it without asking the user to pass it as a parameter?

Reply:
One of the arguments of the callback is  `study`.

Reply:
Agree with @nzw0301, this is the logic I had in mind for this mode.